src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -481,3 +482,28 @@ def delete_nfs_cluster(self, cluster_id):
         except Exception as e:
             log.warning("Failed to delete NFS Cluster")
             return -errno.EINVAL, "", str(e)
+
+    def show_nfs_cluster_info(self, cluster_id):
+        try:
+            self._set_cluster_id(cluster_id)
+            completion = self.mgr.list_daemons(daemon_type='nfs')
+            self.mgr._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            # Here completion.result is a list DaemonDescription objects
+            for cluster in completion.result:
+                if self.cluster_id == cluster.service_id():
+                    """
+                    getaddrinfo sample output: [(<AddressFamily.AF_INET: 2>,
+                    <SocketKind.SOCK_STREAM: 1>, 6, 'xyz', ('172.217.166.98',2049)),
+                    (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '',
+                    ('2404:6800:4009:80d::200e', 2049, 0, 0))]
+                    """
+                    result = {'IP': [ip[4][0] for ip in socket.getaddrinfo(cluster.hostname, 2049,
+                                     flags=socket.AI_CANONNAME, type=socket.SOCK_STREAM)],
+                              'port': 2049 # Default ganesha port
```

Comment:
Does this work right when ganesha is running on a different port?

Reply:
Currently cephadm does not deploy ganesha on non-standard port.

Reply:
Let's think about this. Cephadm doesn't (and hopefully never will) need the raw IP address. On the other hand. in case we want to do enhance cephadm, there is another option:

1. Add a new property `ip` or something similar to

https://github.com/ceph/ceph/blob/c7b45bd1d64e36d7fa15374cf92b20755c2051e5/src/python-common/ceph/deployment/hostspec.py#L7

2. Enhance 

https://github.com/ceph/ceph/blob/af0d8af5e56d03c927a459976019e2d5fb56d9b1/src/pybind/mgr/cephadm/module.py#L1123

to run also `getaddrinfo` for the host. (Do it here, to be able to return the current IP of a host)

4. update the IP in the inventory, similar to

https://github.com/ceph/ceph/blob/7c2ed2a84afa938e2087ccf6b45b386090cf5a32/src/pybind/mgr/cephadm/inventory.py#L55-L58

5. let `ceph orch host ls` also return this attribute

6. change this code here to also call `self.mgr.get_hosts()`

This way, everyone would benefit from this information. 

But I'm perfectly happy with keeping this code here in `show_nfs_cluster_info`

Reply:
> Let's think about this. Cephadm doesn't (and hopefully never will) need the raw IP address.

I imagine what cephadm has for a hostname is just whatever gets passed to `getaddrinfo` (in ssh or elsewhere). Whether that "hostname" is also just a numeric one (i.e. an IP address) is irrelevant to it. (Edit:) Actually, I think it did matter because what I ended up doing in ceph-linode was lookup the FQDN for each VM:

https://github.com/batrick/ceph-linode/blob/cephadm/cephadm.yml#L63-L69

Which does not thrill me. I do think cephadm should be willing to take an IP address for the host so it works with private LANs. IMO, there really should not be any requirement that the cluster have hostnames which can be looked up via DNS.

Reply:
> > Let's think about this. Cephadm doesn't (and hopefully never will) need the raw IP address.
> 
> I imagine what cephadm has for a hostname is just whatever gets passed to `getaddrinfo` (in ssh or elsewhere). Whether that "hostname" is also just a numeric one (i.e. an IP address) is irrelevant to it. (Edit:) Actually, I think it did matter because what I ended up doing in ceph-linode was lookup the FQDN for each VM:
> 
> https://github.com/batrick/ceph-linode/blob/cephadm/cephadm.yml#L63-L69
> 
> Which does not thrill me. I do think cephadm should be willing to take an IP address for the host so it works with private LANs. 

you can do that already:

```bash
ceph orch host add <name> <addr>
# e.g.
ceph orch host add myhostname 1.2.3.4
# or
ceph orch host add myhostname myhostname.domain.tld
```

cephadm will then use the `addr` to connect to the host.


> IMO, there really should not be any requirement that the cluster have hostnames which can be looked up via DNS.

yup. that was the reason we added the `addr` field

Reply:
I would prefer moving this to cephadm and do name resolution only if ip is not specified.